candle-nn/src/activation.rs

```diff
@@ -31,6 +33,7 @@ impl super::Module for Activation {
             Self::Relu2 => xs.relu()?.sqr(),
             Self::Relu6 => xs.clamp(0f32, 6f32),
             Self::Silu => crate::ops::silu(xs),
+            Self::GatedSilu => crate::ops::silu(xs),
```

Comment:
It feels a bit error-prone to add a new activation named `GatedSilu` whereas the behavior is the same as `Silu` and the gating has to be enforced in another way. I think we should instead change the type used in the t5 model to a separate type that handles the `gated-variant`, ideally something like:
```rust
struct ActivationWithOptionalGating {
    gated: bool,
    activation: candle_nn::Activation
}
```
And we use a custom serde decoder that recognizes `gated-silu` and `gated-relu` and set `gated = true` for them and for the other values defer to the serde decoder of `candle-nn::Activation`.

Reply:
I think that's a great idea. I gave it a shot, but I'm new to `serde`.

Please let me know if there's a better way to implement it.